Tests/TensorFlowTests/LossTests.swift

```diff
@@ -67,6 +67,18 @@ final class LossTests: XCTestCase {
 
         assertElementsEqual(expected: expectedGradients, actual: gradients)
     }
+    
+    func testNegativeLogLikelihood() {
+        let predicted = Tensor<Float>(shape: [2, 4], scalars: [1, 2, 3, 4, 5, 6, 7, 8])
+        let expectedGradients = Tensor<Float>(
+            shape: [2, 4],
+            scalars: [-0.125, -0.0625, -0.0416666, -0.03125, -0.025, -0.0208333, -0.017857, -0.015625])
+        let gradients = gradient(
+            at: predicted,
+            in: { negativeLogLikelihood(predicted: $0) })
+        
+        assertElementsEqual(expected: expectedGradients, actual: gradients)
```

Comment:
> Can you also have an expected variable which just checks if the outputs are right? You can add the Assert equal to the same test case.

@Shashi456 Correct me if am wrong, but I believe that is being done here.

Reply:
You are actually checking the gradients here. You could name the loss function `testNegativeLogLikelihoodGrad`.